### PR TITLE
Feat/eject menu spoon

### PR DESCRIPTION
**Pull request by mpjuers**
Implement a drive unmount menu which can alternatively open drives in the Finder.

### CODE REVIEW

File: Source/EjectMenu.spoon/init.lua [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+-- EjectMenu:ejectAll()
+-- Ejects all external drives.
+--
+-- returns
+-- nil
+
+function obj:ejectAll ()
+    hs.osascript.applescript(
+        'tell application "Finder" to eject (every disk whose ejectable is true and local volume is true and free space is not equal to 0)'
```
Comment (gineer01) [inline Source/EjectMenu.spoon/init.lua:80]: Is AppleScript better than hs.fs.volume.eject? This spoon also does ejectAll() but with hs.fs.volume : https://github.com/Hammerspoon/Spoons/pull/177/files

**Comment by zzamboni** [comment]
Funny, I just submitted #177 a few days ago. Not exactly the same, but along the same lines :)

File: Source/EjectMenu.spoon/init.lua (hunk 1 above)
Comment (mpjuers) [reply Source/EjectMenu.spoon/init.lua:80]: I don't know that it's any better. AppleScript has given me the best results in the past for ejecting drives consistently and without intervention, though I haven't tried the `hs.fs.volume.eject`.

**Comment by mpjuers** [comment]
> Funny, I just submitted #177 a few days ago. Not exactly the same, but along the same lines :)

We might be able to merge the two functionality-wise. Yours has some features that mine is missing, notably unmount on sleep. Mine allows you to cmd-click on the menubar items and open the volume in Finder.

**Comment by zzamboni** [comment]
@mpjuers true! I haven't had a chance to look at your code, but I'll try to give it a go during the weekend.

**Comment by mpjuers** [comment]
> @mpjuers true! I haven't had a chance to look at your code, but I'll try to give it a go during the weekend.

Please be aware that I did an interactive rebase and force-pushed to this branch since making the PR.

**Comment by zzamboni** [comment]
A couple of things I noticed:

- I'm getting duplicated entries in the menu and also in the console output from the spoon:
image
```
ArmorATD is ejectable.
Macintosh HD is ejectable.
Macintosh HD is ejectable.
Macintosh HD - Data is ejectable.
Macintosh HD - Data is ejectable.
Macintosh HD - Data is ejectable.
Macintosh HD - Data is ejectable.
```

- The spoon is detecting my Mac's internal partitions as "ejectable". I think this might be because you are checking for the "Ejectable" attribute. It seems better to check the "Internal" flag, as I did in the EjectVolumes spoon: https://github.com/Hammerspoon/Spoons/pull/177/files?file-filters%5B%5D=.lua#diff-dd562edc937238fb8ee9d18f4142dc8fR73

- Instead of using AppleScript to get the volumes, it might be better to use the `hs.volumes.allVolumes()` call: https://github.com/Hammerspoon/Spoons/pull/177/files?file-filters%5B%5D=.lua#diff-dd562edc937238fb8ee9d18f4142dc8fR90

**Comment by mpjuers** [comment]
I've updated with your suggestions. Let me know if this helps. I only sporadically ran into those problems on my machine and I wasn't able to troubleshoot it.

**Comment by zzamboni** [comment]
@mpjuers I have integrated your menu-handling code into #177, bringing the functionality of both spoons into one. If you prefer, I'd be happy to rename the spoon to EjectMenu. I also prefer this name myself instead of EjectVolumes, since I think it better describes the functionality.

Before I rename the spoon - would it be OK with you to close this PR and ask @cmsj to review/merge #177 to have the combined functionality?

**Comment by mpjuers** [comment]
> @mpjuers I have integrated your menu-handling code into #177, bringing the functionality of both spoons into one. If you prefer, I'd be happy to rename the spoon to EjectMenu. I also prefer this name myself instead of EjectVolumes, since I think it better describes the functionality.

Yes, I think that works well.

> Before I rename the spoon - would it be OK with you to close this PR and ask @cmsj to review/merge #177 to have the combined functionality?

Yes, that's fine.

**Comment by zzamboni** [comment]
@mpjuers done!

**Comment by zzamboni** [comment]
@mpjuers should any of the changes you have pushed recently be merged into #177 as well?

**Comment by mpjuers** [comment]
@zzamboni Yes.

**Comment by zzamboni** [comment]
@mpjuers thanks. The error checking was already similar in #177, I added now an "All Ejected" notification when appropriate, as you did. I also fixed today some bugs with the eject_on_sleep and eject_on_lid_close actions.

If the coalesced spoon from #177 works for you, I would propose closing this PR to avoid confusion.

Thanks! I'm happy we managed to join our ideas to create a more functional spoon.

**Comment by mpjuers** [comment]
@zzamboni Yes, let's close.

**Pull request closed without merging**